### PR TITLE
Spdx validate none

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

**Review by fviernau — DISMISSED** [review]
I guess @mnonnenmacher should review this as well.

File: spdx-utils/src/main/kotlin/SpdxExpression.kt [hunk 1]
```diff
@@ -413,11 +413,13 @@ class SpdxLicenseIdExpression(
...
     override fun validate(strictness: Strictness) {
-        when (strictness) {
-            Strictness.ALLOW_ANY -> Unit // Return something non-null.
-            Strictness.ALLOW_DEPRECATED -> spdxLicense
-            Strictness.ALLOW_CURRENT -> spdxLicense?.takeUnless { spdxLicense.deprecated }
-        } ?: throw SpdxException("'$this' is not a valid SPDX license id.")
+        val isValid = when (strictness) {
+            Strictness.ALLOW_ANY -> true
+            Strictness.ALLOW_DEPRECATED -> spdxLicense != null
+            Strictness.ALLOW_CURRENT -> spdxLicense?.takeUnless { spdxLicense.deprecated } != null
```
Comment (mnonnenmacher) [inline spdx-utils/src/main/kotlin/SpdxExpression.kt:419]: Can be simplified to `spdxLicense?.deprecated == false`.

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: spdx-utils/src/main/kotlin/SpdxExpression.kt (hunk 1 above)
Comment (sschuberth) [reply spdx-utils/src/main/kotlin/SpdxExpression.kt:419]: Done.

**Review by mnonnenmacher — APPROVED** [review]

**Comment by sschuberth** [comment]
Ignoring the failed Travis check which was fixed meanwhile in [here](https://github.com/oss-review-toolkit/ort/pull/3310/commits/fcb291c70b6a48f0cf7a1950d9c38050b764d475).

**Pull request merged**